# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1374526684 -7200
#      Mon Jul 22 22:58:04 2013 +0200
# Node ID 1b549a0c3ca43e33bfe38b2f493e4a72ac4c99b1
# Parent  55b76fd1244b3b92c22d37a82fe4bfe26fdbda3f
text.m: Update to use new __plt_get_axis_arg__.

* scripts/plot/text.m: Update to use new __plt_get_axis_arg__.

diff --git a/scripts/plot/text.m b/scripts/plot/text.m
--- a/scripts/plot/text.m
+++ b/scripts/plot/text.m
@@ -113,27 +113,33 @@ function h = text (varargin)
     nt = 1;
   endif
 
   ## Any remaining inputs must occur as PROPERTY/VALUE pairs
   if (rem (numel (varargin), 2) != 0)
     print_usage ();
   endif
 
+  ## Get axis argument which may be in a 'parent' PROP/VAL pair
+  [hax, varargin] = __plt_get_axis_arg__ ("text", varargin{:});
+
+  if (isempty (hax))
+    hax = gca ();
+  endif
+
   if (nx == ny && nx == nz && (nt == nx || nt == 1 || nx == 1))
     pos = [x(:), y(:), z(:)];
-    ca = gca ();
     htmp = zeros (nt, 1);
     if (nx == 1)
-      htmp = __go_text__ (ca, "string", label{1},
+      htmp = __go_text__ (hax, "string", label{1},
                           varargin{:},
                           "position", pos);
     elseif (nx == nt)
       for n = 1:nt
-        htmp(n) = __go_text__ (ca, "string", label{n},
+        htmp(n) = __go_text__ (hax, "string", label{n},
                                varargin{:},
                                "position", pos(n,:));
       endfor
       __request_drawnow__ ();
     else
       error ("text: dimension mismatch for coordinates and LABEL");
     endif
   elseif (nt == nx || nt == 1 || nx == 1)
